.github/workflows/version_check.yml

```diff
@@ -0,0 +1,198 @@
+
+          project_toml = TOML.parsefile(PROJECT_TOML_PATH)
+          project_version_str = project_toml["compat"]["Turing"]
+          project_version = VersionNumber(project_version_str)
+          println("Project.toml version: ", project_version_str)
+
+          manifest_toml = TOML.parsefile(MANIFEST_TOML_PATH)
+          manifest_version = VersionNumber(manifest_toml["deps"]["Turing"][1]["version"])
+          println("Manifest.toml version: ", manifest_version)
+
+          errors = []
+
+          if ENV["TARGET_IS_MASTER"] == "true"
+              # Fetch the latest version from GitHub and update files to match this version
+              # if necessary.
+
+              resp = HTTP.get("https://api.github.com/repos/TuringLang/Turing.jl/releases/latest")
+              latest_version = VersionNumber(JSON.parse(String(resp.body))["tag_name"])
+              println("Latest Turing.jl version: ", latest_version)
+
+              if !major_minor_match(latest_version, project_version)
+                  push!(errors, "$(PROJECT_TOML_PATH) out of date")
+                  println("$(PROJECT_TOML_PATH) is out of date; updating")
+                  update_project_toml(PROJECT_TOML_PATH, latest_version)
+              end
+
+              if !major_minor_match(latest_version, quarto_version)
+                  push!(errors, "$(QUARTO_YML_PATH) out of date")
+                  n_errors += 1
+                  println("$(QUARTO_YML_PATH) is out of date; updating")
+                  update_quarto_yml(QUARTO_YML_PATH, latest_version)
+              end
+
+              if !major_minor_patch_match(latest_version, manifest_version)
+                  push!(errors, "$(MANIFEST_TOML_PATH) out of date")
+                  # Attempt to automatically update Manifest
+                  println("$(MANIFEST_TOML_PATH) is out of date; updating")
+                  old_env = Pkg.project().path
+                  Pkg.activate(".")
+                  Pkg.update()
+                  # Check if versions match now, error if not
+                  Pkg.activate(old_env)
+                  manifest_toml = TOML.parsefile(MANIFEST_TOML_PATH)
+                  manifest_version = VersionNumber(manifest_toml["deps"]["Turing"][1]["version"])
+                  if !major_minor_patch_match(latest_version, manifest_version)
+                      push!(errors, "Failed to update $(MANIFEST_TOML_PATH) to match latest Turing.jl version")
+                  end
+              end
+
+              if isempty(errors)
+                  println("All good")
+              else
+                  error("The following errors occurred during version checking: \n", join(errors, "\n"))
+              end
+
+          else
+              # Don't attempt to fetch the latest version; just check for consistency
+              # and error if versions don't match.
+              if !major_minor_match(quarto_version, project_version, manifest_version)
```

Comment:
Is there a significance to this ignoring `patch` for `manifest_version`, but the earlier if-branch not doing so?

Reply:
Yes, basically, _quarto.yml and Project.toml don't include patch numbers, but the Manifest does. So there's no source of truth for comparison.

On the other branch, we can get the desired patch number from the GitHub API. In principle we could get it here too (by enumerating every tag on GitHub, filtering for the ones with the correct major+minor, then picking the one with the largest patch number). It's a bit of work because to do it correctly we have to handle the result pagination from GitHub's API response, so I didn't bother doing that. Should we?

Reply:
Naah, I think that's fine. Thanks for the explanation.